avm/res/network/private-link-service/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,97 @@
+targetScope = 'subscription'
+
+metadata name = 'WAF-aligned'
+metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-network.privatelinkservices-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'nplswaf'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2022-09-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    virtualNetworkName: 'dep-${namePrefix}-vnet-${serviceShort}'
+    loadBalancerName: 'dep-${namePrefix}-lb-${serviceShort}'
+    location: resourceLocation
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: resourceLocation
+    lock: {
+      kind: 'CanNotDelete'
```

Comment:
Also in this case, please keep waf-aligned test to cover properties specific to waf. In general locks are not required

Reply:
@eriqua is there a specific WAF list i missed from the AVM doc I can use as reference to what constitutes a WAF aligned test case?

Reply:
@eriqua - the current stance is that if a lock exists in WAF Alligned we will not ask for removal. Nor will we ask for addition visa versa!

Reply:
And @ahmadabdalla as per your question - no 👀 

We do mostly rely on PSRule - however, with common sense. That means in some instances we opt to supress a rule if it appears too opinionated (e.g., enforcing managed identities as opposed to just no identity).

Reply:
sorry @ahmadabdalla - We ironed this out today, please remove locks, and any other parameters that do not need to be passed to satify PSRule from the WAF Alligned test

Reply:
@ahmadabdalla sorry, but some details are still being defined as we go. To your point, we should better document what a WAF aligned test case should include, although it may be a gray zone for some properties. I'll open an issue in AVM

Reply:
No probs. @eriqua @ChrisSidebotham I've removed the locks